tests/functional_tests/ffront_tests/builtins_tests/test_gt4py_builtins.py

```diff
@@ -0,0 +1,427 @@
+# -*- coding: utf-8 -*-
+#
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+from collections import namedtuple
+from typing import TypeVar
+
+import numpy as np
+import pytest
+
+from functional.common import DimensionKind
+from functional.ffront.decorator import field_operator, program
+from functional.ffront.fbuiltins import (
+    Dimension,
+    Field,
+    FieldOffset,
+    broadcast,
+    float64,
+    int64,
+    max_over,
+    min_over,
+    neighbor_sum,
+    where,
+)
+from functional.iterator.embedded import (
+    NeighborTableOffsetProvider,
+    index_field,
+    np_as_located_field,
+)
+from functional.program_processors.runners import gtfn_cpu, roundtrip
+
+
+@pytest.fixture(params=[roundtrip.executor, gtfn_cpu.run_gtfn])
+def fieldview_backend(request):
+    yield request.param
+
+
+def debug_itir(tree):
+    """Compare tree snippets while debugging."""
+    from devtools import debug
+
+    from eve.codegen import format_python_source
+    from functional.program_processors import EmbeddedDSL
+
+    debug(format_python_source(EmbeddedDSL.apply(tree)))
+
+
+DimsType = TypeVar("DimsType")
+DType = TypeVar("DType")
+
+IDim = Dimension("IDim")
+JDim = Dimension("JDim")
+
+size = 10
+mask = np_as_located_field(IDim)(np.zeros((size,), dtype=bool))
+mask.array()[size // 2] = True
+a_int = np_as_located_field(IDim)(np.random.randn(size).astype("int64"))
+b_int = np_as_located_field(JDim)(np.random.randn(size).astype("int64"))
+out_int = np_as_located_field(IDim, JDim)(np.zeros((size, size), dtype=int64))
+a_float = np_as_located_field(IDim)(np.random.randn(size).astype("float64"))
+b_float = np_as_located_field(IDim)(np.random.randn(size).astype("float64"))
+out_float = np_as_located_field(IDim)(np.zeros((size), dtype=float64))
```

Comment:
There is as lot of code duplication now in other files, that should be moved to a separate file.